frontend/src/metabase/query_builder/components/QueryModals.tsx

```diff
@@ -0,0 +1,356 @@
+import React from "react";
+import { connect } from "react-redux";
+
+import { t } from "ttag";
+import _ from "underscore";
+
+import Questions from "metabase/entities/questions";
+import { ROOT_COLLECTION } from "metabase/entities/collections";
+
+import { MODAL_TYPES } from "metabase/query_builder/constants";
+
+import Modal from "metabase/components/Modal";
+
+import SaveQuestionModal from "metabase/containers/SaveQuestionModal";
+import QuestionSavedModal from "metabase/components/QuestionSavedModal";
+import AddToDashSelectDashModal from "metabase/containers/AddToDashSelectDashModal";
+
+import CollectionMoveModal from "metabase/containers/CollectionMoveModal";
+import ArchiveQuestionModal from "metabase/questions/containers/ArchiveQuestionModal";
+import QuestionEmbedWidget from "metabase/query_builder/containers/QuestionEmbedWidget";
+
+import { CreateAlertModalContent } from "metabase/query_builder/components/AlertModals";
+import { ImpossibleToCreateModelModal } from "metabase/query_builder/components/ImpossibleToCreateModelModal";
+import NewDatasetModal from "metabase/query_builder/components/NewDatasetModal";
+import EntityCopyModal from "metabase/entities/containers/EntityCopyModal";
+import BulkFilterModal from "metabase/query_builder/components/filters/modals/BulkFilterModal";
+import NewEventModal from "metabase/timelines/questions/containers/NewEventModal";
+import EditEventModal from "metabase/timelines/questions/containers/EditEventModal";
+import MoveEventModal from "metabase/timelines/questions/containers/MoveEventModal";
+import PreviewQueryModal from "metabase/query_builder/components/view/PreviewQueryModal";
+import ConvertQueryModal from "metabase/query_builder/components/view/ConvertQueryModal";
+import QuestionMoveToast from "metabase/questions/components/QuestionMoveToast";
+import { Card, Collection, User } from "metabase-types/api";
+import StructuredQuery from "metabase-lib/queries/StructuredQuery";
+import Question from "metabase-lib/Question";
+
+const mapDispatchToProps = {
+  setQuestionCollection: Questions.actions.setCollection,
+};
+
+type ModalType = typeof MODAL_TYPES[keyof typeof MODAL_TYPES];
+
+interface QueryModalsProps {
+  questionAlerts: any;
+  user: User;
+  modal: ModalType;
+  modalContext: number;
+  question: Question;
+  initialCollectionId: number;
+  updateQuestion: (question: Question, config?: object) => void;
+  setQueryBuilderMode: (mode: string) => void;
+  originalQuestion: Question;
+  card: Card;
+  onCreate: (question: Question, flag?: boolean) => void;
+  onSave: (question: Question, flag?: boolean) => void;
+  onCloseModal: () => void;
+  onOpenModal: (modal: ModalType) => void;
+  onChangeLocation: (location: string) => void;
+  setQuestionCollection: (
+    { id }: Partial<Collection>,
+    collection: Collection,
+    opts: Record<string, unknown>,
+  ) => void;
```

Comment:
I believe the first argument should be `Partial<Card>` (we should make the ID required tho)

Reply:
another great catch, changed it to `Pick<Card, 'id'>`